test/python/sdpa/frost/test_sdpa_sm80_frontend_integration.py

```
@@ -0,0 +1,256 @@
+@_SM80
+def test_bwd_engine_bhsd_contiguous_layout():
+    """Backward counterpart of the dense_flex layout regression."""
+    st = (H * S * D, S * D, D, 1)  # BHSD-contiguous
+    dims = (B, H, S, D)
+    torch.manual_seed(0)
+    bufs = {n: torch.randn(dims, dtype=torch.float16, device="cuda") for n in ("q", "k", "v", "do")}
+
+    # LSE from a matching fwd run (wrapper path; layout-independent).
+    from cudnn.sdpa import sdpa_fwd_wrapper_sm80
+
+    phys = lambda t: t.permute(0, 2, 1, 3).contiguous().permute(0, 2, 1, 3)
```

Ruff (0.16.1)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `phys` as a `def`

(E731)